app/grandchallenge/evaluation/models.py

```diff
@@ -640,7 +638,31 @@ def submission_period_is_open_now(self):
 
     @property
     def open_for_submissions(self):
-        return self.submission_period_is_open_now and self.submission_limit > 0
+        return (
+            self.submission_period_is_open_now
+            and self.submission_limit > 0
+            and not self.submission_limit_reached
+        )
+
+    @property
+    def submission_limit_reached(self):
+        return (
+            self.percent_of_total_submission_limit
+            and self.percent_of_total_submission_limit >= 100
+        )
+
+    @property
+    def percent_of_total_submission_limit(self):
```

Comment:
Looks good, might be worth to rename the properties too now the underlying field has changed name. So `percent_of_total_submissions_allowed` and `exceeds_total_number_of_submissions_allowed` ? Similar things for the 3 new challenge properties. Just thinking that if you don't have `submission_limit` anywhere in this PR the refactoring of the other field becomes easier. What do you think?

Reply:
True, I had actually thought about changing the name to that but then thought it was unnecessarily long - but I see the value in keeping it consistent with the field name to avoid confusion. Will change them.